ibis/expr/api.py

```diff
@@ -678,12 +678,21 @@ def cast(arg, target_type):
     if op.to.equals(arg.type()):
         # noop case if passed type is the same
         return arg
-    else:
-        result = op.to_expr()
-        if not arg.has_name():
-            return result
-        expr_name = 'cast({}, {})'.format(arg.get_name(), op.to)
-        return result.name(expr_name)
+
+    if isinstance(op.to, (dt.Geography, dt.Geometry)):
+        from_geotype = arg.type().geotype
+        to_geotype = op.to.geotype
+        if (
+            from_geotype == to_geotype
+            or (from_geotype is None and to_geotype == 'geometry')
```

Comment:
I think this can be reduced to

```
from_geotype = arg.type().geotype or 'geometry'
to_geotype = op.to.geotype

if from_geotype == to_geotype:
   ...
```

Reply:
looks much better. thanks!  I will change that